src/darwin/frida-helper-backend-glue.m

```diff
@@ -1450,6 +1462,73 @@ static void frida_darwin_helper_backend_launch_using_sbs (NSString * identifier,
   return killed_pid;
 }
 
+static gboolean
+frida_has_active_prewarm (gint pid)
+{
+  gboolean prewarm_active = FALSE;
+  int mib_argmax[] = { CTL_KERN, KERN_ARGMAX };
+  int mib_args[] = { CTL_KERN, KERN_PROCARGS2, 0 };
+  guint8 * buffer;
+  size_t size;
+  gint32 arg_max, argc;
+  guint8 * cursor, * end;
+
+  buffer = NULL;
+  size = sizeof (arg_max);
+
+  arg_max = 0;
+  if (sysctl (mib_argmax, G_N_ELEMENTS (mib_argmax), &arg_max, &size, NULL, 0) != 0)
+    goto beach;
+
+  buffer = g_malloc (arg_max);
+  if (buffer == NULL)
+    goto beach;
+
+  mib_args[2] = pid;
+  size = arg_max;
+
+  if (sysctl (mib_args, G_N_ELEMENTS (mib_args), buffer, &size, NULL, 0) != 0)
+    goto beach;
+  if (size < sizeof (argc))
+    goto beach;
```

Comment:
Is this possible?

Reply:
i didn't observe this but looking at other code bases (chrome) they do it so i thought "better be on the safe side"

Reply:
i can remove and retest

Reply:
double checked in the kernel, and PROCARGS2 always have argc: https://github.com/apple-oss-distributions/xnu/blob/5c2921b07a2480ab43ec66f5b9e41cb872bc554f/bsd/kern/kern_sysctl.c#L1276

(`argc_yes` argument is passed as `1`)